File: 정종길/프로그래머스/카카오 1회독/2022 KAKAO TECH INTERNSHIP/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        HashMap<String, Integer> records = new HashMap<>();

        for(int i = 0; i < choices.length; i ++){
            int choice = choices[i];
            if(choice <= 3){
                String key = "" + survey[i].charAt(0);
                records.put(key, records.getOrDefault(key, 0) + 4 - choice);
            }else if ( 5 <= choice && choice <= 7){
                String key = "" + survey[i].charAt(1);
                records.put(key, records.getOrDefault(key, 0) + choice - 4);
            }
        }

        StringBuilder answer = new StringBuilder();
        answer.append(records.getOrDefault("R", 0) >= records.getOrDefault("T", 0) ? "R" : "T");
        answer.append(records.getOrDefault("C", 0) >= records.getOrDefault("F", 0) ? "C" : "F");
        answer.append(records.getOrDefault("J", 0) >= records.getOrDefault("M", 0) ? "J" : "M");
        answer.append(records.getOrDefault("A", 0) >= records.getOrDefault("N", 0) ? "A" : "N");

        return answer.toString();
    }
}

File: 정종길/프로그래머스/카카오 2회독/2022 KAKAO TECH INTERNSHIP/성격 유형 검사하기.py
from collections import defaultdict

def solution(survey, choices):
    answer = ''
    scores = defaultdict(int)
    
    for i in range(len(survey)):
        c = choices[i]
        q, r = divmod(c, 4)
        if q == 0:
            scores[survey[i][0]] += (4 - r)
        else:
            scores[survey[i][1]] += r
    

    answer += "R" if scores["R"] >= scores["T"] else "T"
    answer += "C" if scores["C"] >= scores["F"] else "F"
    answer += "J" if scores["J"] >= scores["M"] else "M"
    answer += "A" if scores["A"] >= scores["N"] else "N"
    
    return answer